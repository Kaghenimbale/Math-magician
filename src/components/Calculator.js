import React, { useState } from 'react';
import './calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);
  const handleClick = (e) => {
    const answer = calculate({ total, next, operation }, e.target.textContent);
    setTotal(answer.total);
    setNext(answer.next);
    setOperation(answer.operation);
  };
  return (
    <div className="calculator-container">
      <div className="input">
        <p className="values">
          {total}
          {operation}
          {next}
        </p>
      </div>
      <div className="signs">
        <button type="button" onClick={handleClick} className="sign btn1">
          AC
        </button>
        <button type="button" onClick={handleClick} className="sign btn2">
          +/-
        </button>
        <button type="button" onClick={handleClick} className="sign btn3">
          %
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="sign sing-math btn4"
        >
          +
        </button>
        <button type="button" onClick={handleClick} className="sign btn5">
          7
        </button>
        <button type="button" onClick={handleClick} className="sign btn7">
          8
        </button>
        <button type="button" onClick={handleClick} className="sign btn8">
          9
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="sign sing-math btn9"
        >
          x
        </button>
        <button type="button" onClick={handleClick} className="sign btn10">
          4
        </button>
        <button type="button" onClick={handleClick} className="sign btn11">
          5
        </button>
        <button type="button" onClick={handleClick} className="sign btn12">
          6
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="sign sing-math btn13"
        >
          -
        </button>
        <button type="button" onClick={handleClick} className="sign btn14">
          1
        </button>
        <button type="button" onClick={handleClick} className="sign btn15">
          2
        </button>
        <button type="button" onClick={handleClick} className="sign btn16">
          3
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="sign sing-math btn17"
        >
          +
        </button>
        <button type="button" onClick={handleClick} className="sign btn18">
          0
        </button>
        <button type="button" onClick={handleClick} className="sign btn19">
          .
        </button>
        <button
          type="button"
          onClick={handleClick}
          className="sign sing-math btn20"
        >
          =
        </button>
      </div>
    </div>
  );
};

export default Calculator;
