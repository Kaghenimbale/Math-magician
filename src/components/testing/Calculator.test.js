import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Calculator from '../Calculator';

describe('Calculator', () => {
  test('Render Calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should have "Let us do some Math" ', () => {
    render(<Calculator />);
    expect(screen.getByText(/Let us do some Math!/)).toBeInTheDocument();
  });
});
