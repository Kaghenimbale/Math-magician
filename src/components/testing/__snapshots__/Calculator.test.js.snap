// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Render Calculator component 1`] = `
<div
  className="mycalculator"
>
  <div
    className="math-content"
  >
    <h1>
      Let us do some Math!
    </h1>
  </div>
  <div
    className="calculator-container"
  >
    <div
      className="input"
    >
      <p
        className="values"
      >
        0
      </p>
    </div>
    <div
      className="signs"
    >
      <button
        className="sign btn1"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="sign btn2"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="sign btn3"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="sign sing-math btn4"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="sign btn5"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="sign btn7"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="sign btn8"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="sign sing-math btn9"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="sign btn10"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="sign btn11"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="sign btn12"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="sign sing-math btn13"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="sign btn14"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="sign btn15"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="sign btn16"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="sign sing-math btn17"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="sign btn18"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="sign btn19"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="sign sing-math btn20"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
